regnety/models/model.py

```diff
@@ -0,0 +1,90 @@
+import tensorflow as tf
+
+from regnety.regnety.config import get_model_config, ALLOWED_FLOPS
+from regnety.regnety.models.blocks import PreStem, Stem, Stage, Head
+
+from typing import List, Tuple, Union
+
+class RegNetY(tf.keras.Model):
+    """
+    RegNetY model class. Subclassed from tf.keras.Model. Instantiates a randomly
+    initialised model (to be changed after training). User must provide `flops`
+    argument to model.
+
+    Args:
+        flops: Flops of the model eg. "400MF" (Processing one image requires 
+            400 million floating point operations (multiplication, addition))
+        input_shape: A python list or tuple denoting the shape input. Please omit the 
+            batch dimension. eg. (256, 256, 3). Must be greater than or equal to 224.
+    """
+
+    def __init__(self, flops:str = "", input_shape: Union[List, Tuple] = None ):
+        super(RegNetY, self).__init__()
+
+        if flops not in ALLOWED_FLOPS:
+            raise ValueError("`flops` must be one of " + str(ALLOWED_FLOPS))
+        
+        if input_shape is None:
+            self.input_shape = (224,224,3)
+        else: 
+            self.input_shape = input_shape
+        
+        if any([i<224 for i in self.input_shape[:-1]]):
+            raise ValueError('All non-channel dimensions in `input_shape`'
+                             ' must be greater than or equal to 224.')
+
+        try:
+            assert len(self.input_shape) == 3
+        except:
+            raise ValueError('Input shape is invalid. Please enter input shape '
+                             ' as (height, width, 3)')
+
+
+        self.flops = flops
+        self.config = get_model_config(self.flops)
+        self.model = self._get_model_with_config(self.config)
+
+    def build(self, input_shape):
+        self.model.build(input_shape)
```

Comment:
I think we should use the input shape provided or set during initialization here.

Reply:
I don't think we should do that. If the input shape provided by the user is incorrect, then we must rely on the shape of the inputs. We can use this shape for InputLayer. Also, I'm not sure about the behavior if we use the user defined shape.

Reply:
If this is the case, then are we letting the users initialize the model with the argument `input_shape`?

Reply:
Is it okay if we discuss this in tomorrow's meet? I have a couple of questions related to this.